pkg/ccl/changefeedccl/changefeed_test.go

```diff
@@ -7209,6 +7209,19 @@ func TestChangefeedContinuousTelemetryOnTermination(t *testing.T) {
 		seen.Store(false)
 		sqlDB.Exec(t, `INSERT INTO foo VALUES (2)`)
 		sqlDB.Exec(t, `INSERT INTO foo VALUES (3)`)
+
+		s.TestingKnobs.DistSQL.(*execinfra.TestingKnobs).Changefeed.(*TestingKnobs).OverrideContinuousTelemetryLastEmit =
```

Comment:
It might be a data race to modify hooks while there's a running changefeed. Related to my point in https://github.com/cockroachdb/cockroach/pull/144161#issuecomment-2791016850, another option is we could have a hook that's called when the changefeed encounters a transient error that resets the expected logs when that happens.